src/components/axes/timeAxis.ts

```diff
@@ -306,10 +320,20 @@ export module Axis {
         );
     }
 
-    private _cleanTier(index: number) {
-      this._tierLabelContainers[index].selectAll("." + AbstractAxis.TICK_LABEL_CLASS).remove();
-      this._tierMarkContainers[index].selectAll("." + AbstractAxis.TICK_MARK_CLASS).remove();
-      this._tierBaselines[index].style("visibility", "hidden");
+    private _cleanTiers() {
+      var oldTiersNumber = this._tierLabelContainers.length;
+
+      if (! (this._tierLabelContainers.length === this._tierMarkContainers.length &&
+             this._tierLabelContainers.length === this._tierBaselines.length)) {
+        throw Error("Tiers could not be cleaned properly");
+      }
```

Comment:
We currently don't have a clear idea of error catching in Plottable.  I assume that this is similar to an IllegalStateException in Java land, but if this error were to be reported to the user... I don't see too much of how this would help them debug.

I would say remove it for now for consistency and since I'm unsure of how helpful this would be.  If you can think of a good error infrastructure though, we should definitely discuss it in the near future.